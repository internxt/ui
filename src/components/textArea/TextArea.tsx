const TextArea = ({
  disabled = false,
  accentColor,
  placeholder = '',
}: {
  disabled?: boolean;
  accentColor?: 'red';
  placeholder?: string;
}): JSX.Element => {
  return (
    <textarea
      disabled={disabled}
      placeholder={placeholder}
      className={`
        w-full h-full py-4 px-3.5 bg-transparent border rounded-md outline-none text-lg font-regular resize-none
        placeholder:text-gray-30
        ${!disabled ? 'border-gray-20 text-gray-100' : 'border-gray-5 text-gray-40'}
        ${!accentColor && 'border-gray-20 focus:border-primary focus:ring focus:ring-primary/10'}
        ${accentColor === 'red' && 'border-red focus:ring focus:ring-red/10'}
  `}
    />
  );
};

export default TextArea;
