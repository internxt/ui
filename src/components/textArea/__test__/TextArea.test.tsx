import React from 'react';
import { describe, expect, it } from 'vitest';
import TextArea from '../TextArea';
import renderer from 'react-test-renderer';

describe('Button component', () => {
  it('TextArea enabled should render correctly', () => {
    const button = renderer.create(<TextArea />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('TextArea enabled with placeholder should render correctly', () => {
    const button = renderer.create(<TextArea placeholder="Placeholder" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('TextArea enabled with accent color red should render correctly', () => {
    const button = renderer.create(<TextArea accentColor="red" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('TextArea disabled should render correctly', () => {
    const button = renderer.create(<TextArea disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
