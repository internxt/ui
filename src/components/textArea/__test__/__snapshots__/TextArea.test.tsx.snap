// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > TextArea disabled should render correctly 1`] = `
<textarea
  className="
        w-full h-full py-4 px-3.5 bg-transparent border rounded-md outline-none text-lg font-regular resize-none
        placeholder:text-gray-30
        border-gray-5 text-gray-40
        border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
        false
  "
  disabled={true}
  placeholder=""
/>
`;

exports[`Button component > TextArea enabled should render correctly 1`] = `
<textarea
  className="
        w-full h-full py-4 px-3.5 bg-transparent border rounded-md outline-none text-lg font-regular resize-none
        placeholder:text-gray-30
        border-gray-20 text-gray-100
        border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
        false
  "
  disabled={false}
  placeholder=""
/>
`;

exports[`Button component > TextArea enabled with accent color red should render correctly 1`] = `
<textarea
  className="
        w-full h-full py-4 px-3.5 bg-transparent border rounded-md outline-none text-lg font-regular resize-none
        placeholder:text-gray-30
        border-gray-20 text-gray-100
        false
        border-red focus:ring focus:ring-red/10
  "
  disabled={false}
  placeholder=""
/>
`;

exports[`Button component > TextArea enabled with placeholder should render correctly 1`] = `
<textarea
  className="
        w-full h-full py-4 px-3.5 bg-transparent border rounded-md outline-none text-lg font-regular resize-none
        placeholder:text-gray-30
        border-gray-20 text-gray-100
        border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
        false
  "
  disabled={false}
  placeholder="Placeholder"
/>
`;
